packages/client/hmi-client/src/components/model/variables/tera-transitions.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+	<tera-variables
+		:variable-list="transitionsList"
+		:disabled-inputs="['concept', 'description', 'name', 'unitExpression']"
+		show-matrix
+		@open-matrix="(id: string) => (matrixModalId = id)"
+		@update-variable="$emit('update-transition', $event)"
+	/>
+	<teleport to="body">
+		<tera-stratified-matrix-modal
+			v-if="matrixModalId"
+			:id="matrixModalId"
+			:mmt="mmt"
+			:mmt-params="mmtParams"
+			:stratified-matrix-type="StratifiedMatrix.Rates"
+			is-read-only
+			@close-modal="matrixModalId = ''"
+		/>
+	</teleport>
+</template>
+
+<script setup lang="ts">
+import { ref, computed } from 'vue';
+import { StratifiedMatrix } from '@/types/Model';
+import type { ModelVariable } from '@/types/Model';
+// import type { Transition } from '@/types/Types';
+import type { MiraModel, MiraTemplateParams } from '@/model-representation/mira/mira-common';
+import { collapseTemplates } from '@/model-representation/mira/mira';
+import TeraVariables from '@/components/model/variables/tera-variables.vue';
+import TeraStratifiedMatrixModal from '@/components/model/petrinet/model-configurations/tera-stratified-matrix-modal.vue';
+
+const props = defineProps<{
+	mmt: MiraModel;
+	mmtParams: MiraTemplateParams;
+	transitions: any[]; // Support for Transition type needs work, missing expression, description and name
+}>();
+
+defineEmits(['update-transition']);
+
+const collapsedTemplates = computed(() => collapseTemplates(props.mmt));
+
+const transitionsList = computed<
+	{
+		base: ModelVariable;
+		children: ModelVariable[];
+		isParent: boolean;
+	}[]
+>(() =>
+	Array.from(collapsedTemplates.value.matrixMap.keys())
+		.flat()
+		.map((templateId) => {
```

Comment:
What is the point of the Array and flat here? Would it be simpler to directly
```
[...collapsedTemplates.value.matrixMap.keys()].map((templateId) => { ......
```

Reply:
Or better a flatMap()

Reply:
I actually gotta check why it's like this, I forgot

Reply:
> Or better a flatMap()

`keys()` are already flat

Reply:
removed them, looks like I blindly copied from `tera-initial-table` and `tera-parameter-table` (which I fixed too)